tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -3794,29 +3590,31 @@ Status FragmentMetadata::load_footer(
   if (loaded_metadata_.footer_)
     return Status::Ok();
 
-  Buffer buff;
-  shared_ptr<ConstBuffer> cbuff = nullptr;
-  if (f_buff == nullptr) {
+  Tile tile;
+  if (f_tile == nullptr) {
     has_consolidated_footer_ = false;
-    RETURN_NOT_OK(read_file_footer(&buff, &footer_offset_, &footer_size_));
-    cbuff = make_shared<ConstBuffer>(HERE(), &buff);
+    RETURN_NOT_OK(read_file_footer(&tile, &footer_offset_, &footer_size_));
+
+    f_tile = &tile;
+    offset = 0;
   } else {
-    footer_size_ = 0;
+    footer_size_ = 0; // adjusted at end of routine based on buffer consumed
     footer_offset_ = offset;
     has_consolidated_footer_ = true;
-    cbuff = make_shared<ConstBuffer>(HERE(), f_buff);
-    cbuff->set_offset(offset);
   }
+  Deserializer deserializer(static_cast<uint8_t*>(f_tile->data()) + offset, f_tile->size() - offset);
```

Comment:
`static_cast<uint8_t*>(f_tile->data())` can be converted to: `f_tile->data_as<uint8_t>()`

Reply:
changed